programs/util.h

```diff
@@ -84,7 +84,9 @@ extern "C" {
 /*-**************************************************************
 *  Basic Types
 *****************************************************************/
-#if  !defined (__VMS) && (defined (__cplusplus) || (defined (__STDC_VERSION__) && (__STDC_VERSION__ >= 199901L) /* C99 */) )
+#ifndef BASIC_TYPES_DEFINED
```

Comment:
The initial intention was to `#include "mem.h"` to access these definitions.

Why do you prefer to repeat them here, implying the need of an additional macro to avoid double-definitions ?